core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -1,22 +1,214 @@
 package com.databricks.labs.remorph.parsers.tsql
 
-import org.antlr.v4.runtime.Token
-import org.antlr.v4.runtime.tree.TerminalNode
-
-import com.databricks.labs.remorph.parsers.intermediate.{Column, Literal}
+import com.databricks.labs.remorph.parsers.intermediate.{Column, Expression, Literal}
 import com.databricks.labs.remorph.parsers.tsql.TSqlParser._
 import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
+import org.antlr.v4.runtime.Token
+import org.antlr.v4.runtime.tree.TerminalNode
 
+import java.util.Locale
 import scala.collection.JavaConverters._
 
 class TSqlExpressionBuilder
     extends TSqlParserBaseVisitor[ir.Expression]
     with ParserCommon
     with IncompleteParser[ir.Expression] {
 
+  case class FunDef(argMin: Int, argMax: Int = Int.MaxValue, convertible: Boolean = true)
+
   protected override def wrapUnresolvedInput(unparsedInput: String): ir.UnresolvedExpression =
     ir.UnresolvedExpression(unparsedInput)
 
+  // TODO: Add more functions as we find them
+  private val functions: Map[String, FunDef] = Map(
```

Comment:
Yay tabulated function! Brings up memories of game programming in C++ ❤️ 

One minor drawback is that it trumps code coverage somehow (all the lines will be considered as covered as long as the `function` member is evaluated in a test, which happens whenever a `TSqlExpressionBuilder` object is instantiated). 

Good news is, the changes required to fix that are fairly mechanical:
```suggestion
  def functionArities(functionName: String): FunDef = functionName match {
```
and then rewrite every following `"FUNC_NAME" -> FuncDef(x, y)` line as `case "FUNC_NAME" => FuncDef(x, y)`.

Because doing so would make a lot of lines marked as uncovered, we should move the definition to a separate file (for example as a member of a `FunctionArities` object) and decide that having uncovered lines in that file is acceptable for now. 

It would also make it easier to later adapt and reuse this information in other dialects.

Reply:
Yeah - did not consider it important to have every entry in the table checked. I think then, that moving the function lookup and table to a common file and over time annotating common functions vs TSQL specific vs Sonwflake vs X, is a good idea so I will do that in this PR.